File: backend/src/main/java/com/project1/backend/model/Product.java
package com.project1.backend.model;

import jakarta.persistence.*;
import jdk.jfr.DataAmount;
import lombok.Data;

import java.time.LocalDate;

@Entity
@Table(name = "products")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;
    private double price;
    private int quantity;
    private String category;
    private LocalDate dateOfCreation;
    private String description;

}

File: frontend/src/app/model/Product.ts
import {Category} from './Category';
import {Supplier} from './Supplier';


export interface Product {
  id?: number;
  name: string;
  price: number;
  quantity: number;
  category: Category;
  dateOfCreation: Date;
  supplier: Supplier;
  description: string;
}
